### PR TITLE
CMCL-0000: couple of cleanups

### PR DESCRIPTION
**Pull request by gaborkb**
Related to PR: https://github.com/Unity-Technologies/com.unity.cinemachine/pull/509/files#diff-b4fad7af055b5617dabcd9f88a4ee7ad8218ba2ecb061a8b0fb803114edd718aR227-R242

- Simplified ThirdPersonFollowEditor handle code
- Fixed a couple of typos (it's simpler/faster to fix them right away than to note them in PR review)
- Changed a couple of types to vars
- Array.Empty instead of [0]

### CODE REVIEW

File: com.unity.cinemachine/Runtime/Behaviours/CinemachineStateDrivenCamera.cs [hunk 1]
```diff
@@ -257,7 +257,7 @@ struct HashPair { public int parentHash; public int hash; }
         int LookupFakeHash(int parentHash, AnimationClip clip)
         {
             if (m_HashCache == null)
-                m_HashCache = new();
+                m_HashCache = new Dictionary<AnimationClip, List<HashPair>>();
```
Comment (glabute) [inline com.unity.cinemachine/Runtime/Behaviours/CinemachineStateDrivenCamera.cs:260]: Why did you change this?

**Review by glabute — APPROVED** [review]

File: com.unity.cinemachine/Runtime/Behaviours/CinemachineStateDrivenCamera.cs (hunk 1 above)
Comment (gaborkb) [reply com.unity.cinemachine/Runtime/Behaviours/CinemachineStateDrivenCamera.cs:260]: I felt it's more clear/readable this way
